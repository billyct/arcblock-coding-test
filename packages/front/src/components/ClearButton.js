import cn from 'classnames'
import {ReactComponent as CloseIcon} from '../assets/close.svg'

const ClearButton = ({className, ...props}) => (
  <button
    className={cn('absolute top-0 bottom-0 right-4 text-gray-500 outline-none focus:outline-none', className)}
    {...props}
  >
    <CloseIcon width={12} height={12}/>
  </button>
)

export default ClearButton
