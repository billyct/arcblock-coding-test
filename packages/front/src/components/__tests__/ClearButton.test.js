import {render} from '@testing-library/react'

import ClearButton from '../ClearButton'

describe.each([
  [
    {}
  ],
  [
    {className: 'a_classname'}
  ]
])('ClearButton Component with %o', attribute => {
  it('should render correct', function () {
    const {container} =render(<ClearButton {...attribute}/>)
    expect(container).toMatchSnapshot()
  })
})
