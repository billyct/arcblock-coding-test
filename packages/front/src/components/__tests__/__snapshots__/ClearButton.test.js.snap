// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButton Component with { className: 'a_classname' } should render correct 1`] = `
<div>
  <button
    class="absolute top-0 bottom-0 right-4 text-gray-500 outline-none focus:outline-none a_classname"
  >
    <svg
      height="12"
      width="12"
    >
      close.svg
    </svg>
  </button>
</div>
`;

exports[`ClearButton Component with {} should render correct 1`] = `
<div>
  <button
    class="absolute top-0 bottom-0 right-4 text-gray-500 outline-none focus:outline-none"
  >
    <svg
      height="12"
      width="12"
    >
      close.svg
    </svg>
  </button>
</div>
`;
